test/it/negative.spec.js

```diff
@@ -0,0 +1,15 @@
+const compiler = require("../utils/compiler");
+const testRoots = require("../utils/tests-roots");
+const negativeRoots = testRoots.getNegativeRoots();
+
+describe("typescript - negative scenarios", () => {
+  it.each(negativeRoots)(
+    "should NOT complie successfully %s folder",
+    async rootPath => {
+      const configurationFilename = `${testRoots.NEGATIVE_ROOT_PATH}/${rootPath}/tsconfig.json`;
+      const result = compiler(configurationFilename);
```

Comment:
i think it would br nicer if in case of an error, the compiler function will throw with the relevant errors.
if you prefer the current api, let's at least rename "result" to "errors"